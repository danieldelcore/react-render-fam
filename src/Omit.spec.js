import React from 'react';
import renderer from 'react-test-renderer';
import { Omit } from '.';

test('Renders the content if predicate is true', () => {
    const data = [
        { id: 1, value: 0 },
        { id: 2, value: 1 },
        { id: 3, value: 5 },
        { id: 4, value: 6 },
        { id: 5, value: 2000 },
    ];

    const component = renderer.create(
        <Omit
            values={data}
            predicates={({ value }) => value > 5}
            render={({ id, value }) => (
                <p key={id}>{value}</p>
            )}
        />
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});

test('Renders the content if all predicates are true', () => {
    const data = [
        { id: 1, value: 0 },
        { id: 2, value: 1 },
        { id: 3, value: 5 },
        { id: 4, value: 6 },
        { id: 5, value: 99 },
        { id: 6, value: 100 },
        { id: 7, value: 2000 },
    ];

    const greaterThanFive = ({ value }) => value > 5;
    const lessThanOneHundred = ({ value }) => value < 100;

    const component = renderer.create(
        <Omit
            values={data}
            predicates={[
                greaterThanFive,
                lessThanOneHundred,
            ]}
            render={({ id, value }) => (
                <p key={id}>{value}</p>
            )}
        />
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});

test('Renders all items if predicates are omitted', () => {
    const data = [
        { id: 1, value: 0 },
        { id: 2, value: 1 },
        { id: 3, value: 5 },
    ];

    const component = renderer.create(
        <Omit
            values={data}
            render={({ id, value }) => (
                <p key={id}>{value}</p>
            )}
        />
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});
