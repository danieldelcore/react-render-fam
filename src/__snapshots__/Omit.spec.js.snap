// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders all items if predicates are omitted 1`] = `
Array [
  <p>
    0
  </p>,
  <p>
    1
  </p>,
  <p>
    5
  </p>,
]
`;

exports[`Renders the content if all predicates are true 1`] = `
Array [
  <p>
    6
  </p>,
  <p>
    99
  </p>,
]
`;

exports[`Renders the content if predicate is true 1`] = `
Array [
  <p>
    6
  </p>,
  <p>
    2000
  </p>,
]
`;
